core/controllers/learner_dashboard_test.py

```diff
@@ -301,10 +299,35 @@ def test_get_message_summaries(self):
         messages_summary = response_dict['message_summary_list']
 
         # Check the summary of the second message.
-        second_message = messages_summary[1]
+        second_message = messages_summary[-1]
```

Comment:
second_message should be the element at index 1 in the array right? Why this change?

Reply:
We changed this so that it would always access the last element rather than the 2nd element in the array. The existing method is not very generalizable, and looking at the code it appears that the person who created this test wants to check the data of the last message rather than the second message.

 For example, if someone made another test and added another message before this message is added, then this test would incorrectly fail. In retrospect, I should have also renamed the variable to last_message. Please advise if this rationale is correct. 

Also regarding the screenshot of the coverage report, I will run it and provide it to you shortly.

Reply:
There seems to be something missing here. So the test does not ensure that on adding 2 threads (There were 2 put requests above), that only 2 threads were added. This seems to be something missing in the existing test. So probably, You can add a check to make sure that there are exactly 2 elements in the list, and then checking for either -1 or 1 would mean the same thing :). 

The idea is that tests should be deterministic, and we should be able to say what should have happened till a point correctly, and assert that the values are as expected. So if the test does fail, then possibly, this test would also need to be edited to convey the change. This will make sure everything is deterministic. 

By changing it to -1, we aren't sure of how many threads are in the list, and we have to be sure there!

Reply:
Done